mount.go

```diff
@@ -187,9 +188,19 @@ type storageHandler func(storage pb.Storage, s *sandbox) (string, error)
 
 // storageHandlerList lists the supported drivers.
 var storageHandlerList = map[string]storageHandler{
-	driver9pType:   virtio9pStorageHandler,
-	driverBlkType:  virtioBlkStorageHandler,
-	driverSCSIType: virtioSCSIStorageHandler,
+	driver9pType:        virtio9pStorageHandler,
+	driverBlkType:       virtioBlkStorageHandler,
+	driverSCSIType:      virtioSCSIStorageHandler,
+	driverEphemeralType: ephemeralStorageHandler,
+}
+
+func ephemeralStorageHandler(storage pb.Storage, s *sandbox) (string, error) {
+	if _, err := os.Stat(storage.MountPoint); os.IsNotExist(err) {
+		os.MkdirAll(storage.MountPoint, os.ModePerm)
```

Comment:
Good catch @sboeuf!

Reply:
@harche You do not even need to check if the directory exists before calling `MkdirAll`.  From the docs,  "If path is already a directory, MkdirAll does nothing and returns nil."

Reply:
@amshinde Well I think you need to check this because in case it does exist, you don't need to call into `commonStorageHandler()`.

Reply:
and why are we not calling then `commonStorageHandler()` then? It may happen that the directory exists, but we skip mounting altogether.

Reply:
@amshinde the tmpfs mount point has to be mounted on the guest only once. Then rest of the containers will bind mount that directory. 

e.g. If all the containers need to have an ephemeral volume in the pod they will bind mount, say /sharedMount directory on the VM. But before that this /sharedMount has to be mounted on the VM backed by tmpfs (but only once)

The above code is tested with a k8s yaml with 3 containers sharing the mount point. 

```
apiVersion: v1
kind: Pod
metadata:
  name: myapp-pod
  labels:
    app: myapp
spec:
  containers:
  # Application Container
  - name: myapp-container
    volumeMounts:
    - mountPath: /cache
      name: cache-volume
    imagePullPolicy: Never
    image: app_image:latest
    command: ['sh', '-c', 'cat /cache/test && sleep 3600']
  initContainers:
  # Copy Data container
  - name: data-container
    volumeMounts:
    - mountPath: /cache
      name: cache-volume
    imagePullPolicy: Never
    image: data_image:latest
    command: ['sh', '-c', 'cp /test /cache/']
  # Append Data container
  - name: init-append-data
    volumeMounts:
    - mountPath: /cache
      name: cache-volume
    imagePullPolicy: Never
    image: test_image:latest
    command: ['sh', '-c', 'echo test >> /cache/test']
  volumes:
  - name: cache-volume
    emptyDir:
      medium: "Memory"
``

Reply:
@amshinde It's is intentionally skipped mounting if dir exists because then every single container end up pointing towards a seperate tmpfs volume and they don't end up sharing the volume. So only once (for the first time) you have to make sure the dir exists and then mount it on guest backed by tmpfs. The corresponding enteries in all other container's OCI spec already point towards bind mounting that dir.

Reply:
@harche Thanks for the explanation, my question was based on the assumption that you would be adding storage just once for the first container.